FWCore/ParameterSet/interface/ParameterSetDescription.h

```diff
@@ -93,45 +97,63 @@ namespace edm {
 
     template <typename T, typename U>
     ParameterDescriptionBase* add(U const& iLabel, T const& value) {
-      return add<T, U>(iLabel, value, true, false, true);
+      return add<T, U>(iLabel, value, true, Modifier::kNone, true);
     }
 
     template <typename T, typename U>
     ParameterDescriptionBase* addUntracked(U const& iLabel, T const& value) {
-      return add<T, U>(iLabel, value, false, false, true);
+      return add<T, U>(iLabel, value, false, Modifier::kNone, true);
     }
 
     template <typename T, typename U>
     ParameterDescriptionBase* addOptional(U const& iLabel, T const& value) {
-      return add<T, U>(iLabel, value, true, true, true);
+      return add<T, U>(iLabel, value, true, Modifier::kOptional, true);
     }
 
     template <typename T, typename U>
     ParameterDescriptionBase* addOptionalUntracked(U const& iLabel, T const& value) {
-      return add<T, U>(iLabel, value, false, true, true);
+      return add<T, U>(iLabel, value, false, Modifier::kOptional, true);
     }
 
     // For the next 4 functions, there is no default so they will not get injected
     // during validation if missing and they will not get written into cfi files.
 
     template <typename T, typename U>
     ParameterDescriptionBase* add(U const& iLabel) {
-      return add<T, U>(iLabel, true, false, true);
+      return add<T, U>(iLabel, true, Modifier::kNone, true);
     }
 
     template <typename T, typename U>
     ParameterDescriptionBase* addUntracked(U const& iLabel) {
-      return add<T, U>(iLabel, false, false, true);
+      return add<T, U>(iLabel, false, Modifier::kNone, true);
     }
 
     template <typename T, typename U>
     ParameterDescriptionBase* addOptional(U const& iLabel) {
-      return add<T, U>(iLabel, true, true, true);
+      return add<T, U>(iLabel, true, Modifier::kOptional, true);
     }
 
     template <typename T, typename U>
     ParameterDescriptionBase* addOptionalUntracked(U const& iLabel) {
-      return add<T, U>(iLabel, false, true, true);
+      return add<T, U>(iLabel, false, Modifier::kOptional, true);
+    }
+
+    template <typename T, typename U>
+    ParameterDescriptionBase* addObsolete(U const& iLabel) {
+      if constexpr (std::is_same_v<T, edm::ParameterSetDescription>) {
```

Comment:
I'm wondering if a short comment here (and below for the untracked) explaining why `ParameterSetDescription` needs a different treatment would be useful for a future reader?